test/support/test_public_website.py

```diff
@@ -23,39 +23,138 @@ def __getitem__(self, item):
 
 class TestPublicWebsite(TestCase):
 
-    def test_reset_password_url(self):
-
-        config = MockConfig()
+    def test_reset_password_url_excludes_port_80_when_scheme_is_http(self):
+        # Given config contains SCHEME 'http and PORT 1234
+        config = MockConfig(scheme='http', port=1234)
         unit = PublicWebsite(config)
 
-        expected_url_substring = "http://mockhost:1234/passwords/reset-password/"
+        # When the reset password url is constructed
         actual_url = unit.reset_password_url('test@email.com')
 
+        # Then it contains the port number
+        expected_url_substring = "http://mockhost:1234/passwords/reset-password/"
         # TODO: ought to be possible to mock the token generator, so we can predict the entire url
         self.assertIn(expected_url_substring, actual_url)
 
-        config = MockConfig(port=80)
+        # Given config contains SCHEME 'http' and PORT 80
+        config = MockConfig(scheme='http', port=80)
         unit = PublicWebsite(config)
 
+        # When the reset password url is constructed
+        actual_url = unit.reset_password_url('test@email.com')
+
         expected_url_substring = "http://mockhost/passwords/reset-password/"
+        # Then it does not contain the port number
+        self.assertIn(expected_url_substring, actual_url)
+
+        # Given config contains SCHEME 'http' and PORT '80' (in string format)
+        config = MockConfig(port='80')
+        unit = PublicWebsite(config)
+
+        # When the reset password url is constructed
+        actual_url = unit.reset_password_url('test@email.com')
+
+        # Then it does not contain the port number
+        expected_url_substring = "http://mockhost/passwords/reset-password/"
+        self.assertIn(expected_url_substring, actual_url)
+
+    def test_reset_password_url_excludes_port_443_when_scheme_is_https(self):
+        # Given config contains SCHEME 'http' and PORT 443
```

Comment:
The comments here say `http` instead of `https`

Reply:
One of the drawbacks of non-executable Gherkin-like syntax :)